pandas/core/internals/construction.py

```diff
@@ -357,7 +370,21 @@ def dict_to_mgr(
         arrays = [
             arr if not is_datetime64tz_dtype(arr) else arr.copy() for arr in arrays
         ]
-    return arrays_to_mgr(arrays, data_names, index, columns, dtype=dtype, typ=typ)
+
+    if copy:
+        # arrays_to_mgr (via form_blocks) won't make copies for EAs
+        # dtype attr check to exclude EADtype-castable strs
+        arrays = [
+            x
+            if not hasattr(x, "dtype") or not isinstance(x.dtype, ExtensionDtype)
+            else x.copy()
+            for x in arrays
+        ]
+        # TODO: can we get rid of the dt64tz special case above?
```

Comment:
This is a change in behaviour, right? So that also EAs in dicts get copied (fixing the original report of https://github.com/pandas-dev/pandas/issues/32960) 

If so, can you add a whatsnew note about this?

Reply:
this isnt intended to be a behavior change, but to take the place of copies that currently happen elsewhere in the status quo (not sure where off the top of my head).  removing this breaks 230 tests.

Reply:
all of them are test_dict_nocopy; the EA inputs dont get copied without this

Reply:
Yes, but right now, constructing a DataFrame from a dict with EAs doesn't copy the EAs. Which is changed with this PR, I assume? (based on the above code, and on your comment "the EA inputs dont get copied without this").I think this is a good change (to ensure consistency now in how a dict gets handled), but it's still a change worth noting if the above is correct.

Reply:
thanks for clarifying.  amended whatsnew